In [4]:
import pandas as pd
dataset=pd.read_csv("train.csv")
dataset.head(2)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

   Parch     Ticket     Fare Cabin Embarked  
0      0  A/5 21171   7.2500   NaN        S  
1      0   PC 17599  71.2833   C85        C

In [5]:
dataset.count()

PassengerId    889
Survived       889
Pclass         889
Name           889
Sex            889
Age            889
SibSp          889
Parch          889
Ticket         889
Fare           889
Cabin          202
Embarked       889
dtype: int64

In [7]:
dataset.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [8]:
NB_dataset=dataset.drop(["Name","Cabin","Ticket"], axis=1)

In [9]:
NB_dataset.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch',
       'Fare', 'Embarked'],
      dtype='object')

In [14]:
from sklearn import preprocessing
from sklearn.model_selection import train_test_split 
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix

In [15]:
le=preprocessing.LabelEncoder()

In [16]:
le.fit(NB_dataset["Sex"])

LabelEncoder()

In [18]:
print(le.classes_)

['female' 'male']


In [30]:
NB_dataset["Sex"]=le.transform(NB_dataset["Sex"])

In [20]:
le.fit(NB_dataset["Sex"])

LabelEncoder()

In [31]:
print(le.classes_)

[0 1]


In [36]:
le.fit(NB_dataset["Embarked"])

LabelEncoder()

In [37]:
NB_dataset["Embarked"]=le.transform(NB_dataset["Embarked"])

In [38]:
print(le.classes_)

[0 1 2]


In [39]:
y=NB_dataset["Survived"]
X=NB_dataset.drop(["Survived","PassengerId"],axis=1)

In [40]:
X.columns

Index(['Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Embarked'], dtype='object')

In [41]:
y.count()

889

In [42]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)

In [43]:
X_train.head(2)

Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
350       1    1  45.0      0      0  35.0000         2
124       3    1  12.0      1      0  11.2417         0

In [44]:
from sklearn.naive_bayes import *

In [46]:
clf=BernoulliNB()

In [47]:
y_pred=clf.fit(X_train,y_train).predict(X_test)

In [53]:
print("The accuracy score of the model is:", accuracy_score(y_test,y_pred,normalize=True)*100)

The accuracy score of the model is: 77.15355805243446


In [49]:
confusion_matrix(y_test,y_pred)

array([[131,  26],
       [ 35,  75]], dtype=int64)

In [52]:
NB_dataset.head(2)

PassengerId  Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0            1         0       3    1  22.0      1      0   7.2500         2
1            2         1       1    0  38.0      1      0  71.2833         0

# Thus in above Naive Bayes Classification Theory implemnetation analysis folling are the inferences worth to be drawn - 

1. When teh Dependent Variable is "Survived" and rest of the Columns data from the dataset(NB_dataset) is taken as Independent Variables the accuracy of the model comes out to be - 77.15355805243446%.
2. Confussion Matrix demonstrate the same accuracy. 

In [74]:
from sklearn.model_selection import train_test_split 
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix
from sklearn.naive_bayes import *
def Naive_Bayes(X,y):
    clf=BernoulliNB()
    X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)
    y_pred=clf.fit(X_train,y_train).predict(X_test)
    print("The accuracy score of the model is:\n", accuracy_score(y_test,y_pred,normalize=True)*100)
    print("The confusion Matrix is:\n", confusion_matrix(y_test,y_pred))

In [75]:
print("The Accuracy_score & Confusion Matrix when DV is Pclass and Rest is IDV is as mentioned below:")
Naive_Bayes((NB_dataset.drop(["Pclass","PassengerId"],axis=1)),NB_dataset["Pclass"])

The Accuracy_score & Confusion Matrix when DV is Pclass and Rest is IDV is as mentioned below:
The accuracy score of the model is:
 59.176029962546814
The confusion Matrix is:
 [[ 30   8  32]
 [  9  10  30]
 [ 24   6 118]]


In [76]:
print("The Accuracy_score & Confusion Matrix when DV is Sex and Rest is IDV is as mentioned below:")
Naive_Bayes((NB_dataset.drop(["Sex","PassengerId"],axis=1)),NB_dataset["Sex"])

The Accuracy_score & Confusion Matrix when DV is Sex and Rest is IDV is as mentioned below:
The accuracy score of the model is:
 74.53183520599251
The confusion Matrix is:
 [[ 49  49]
 [ 19 150]]


In [78]:
print("The Accuracy_score & Confusion Matrix when DV is SibSp and Rest is IDV is as mentioned below:")
Naive_Bayes((NB_dataset.drop(["SibSp","PassengerId"],axis=1)),NB_dataset["SibSp"])

The Accuracy_score & Confusion Matrix when DV is SibSp and Rest is IDV is as mentioned below:
The accuracy score of the model is:
 68.91385767790263
The confusion Matrix is:
 [[162  20   0   0   0   0   0]
 [ 43  22   0   0   0   0   0]
 [  6   2   0   0   0   0   0]
 [  5   2   0   0   0   0   0]
 [  2   0   0   0   0   0   0]
 [  1   0   0   0   0   0   0]
 [  2   0   0   0   0   0   0]]


In [79]:
print("The Accuracy_score & Confusion Matrix when DV is Parch and Rest is IDV is as mentioned below:")
Naive_Bayes((NB_dataset.drop(["Parch","PassengerId"],axis=1)),NB_dataset["Parch"])

The Accuracy_score & Confusion Matrix when DV is Parch and Rest is IDV is as mentioned below:
The accuracy score of the model is:
 71.53558052434457
The confusion Matrix is:
 [[182  16   0   0   0]
 [ 31   9   0   0   0]
 [ 24   3   0   0   0]
 [  1   0   0   0   0]
 [  1   0   0   0   0]]


In [80]:
print("The Accuracy_score & Confusion Matrix when DV is Embarked and Rest is IDV is as mentioned below:")
Naive_Bayes((NB_dataset.drop(["Embarked","PassengerId"],axis=1)),NB_dataset["Embarked"])

The Accuracy_score & Confusion Matrix when DV is Embarked and Rest is IDV is as mentioned below:
The accuracy score of the model is:
 73.40823970037454
The confusion Matrix is:
 [[  0   0  49]
 [  0   0  22]
 [  0   0 196]]


# Thus, We have analysed and developed Naive Bayes Classification Algorithm for the given dataset considering all aspects as Dependent Variable.